### PR TITLE
extractors: fix pornhub

### LINKED ISSUE #663: Bug:下载ph站时发生错误
Console output
```
annie -d 'https://cn.pornhub.com/view_video.php?viewkey=ph5e35bb9dac481'

annie: version 0.9.8, A fast, simple and clean video downloader.


URL:         https://cn.pornhub.com/view_video.php?viewkey=ph5e35bb9dac481
Method:      GET
Headers:     http.Header{
    "User-Agent":      {"Mozilla/5.0 (Macintosh; Intel Mac OS X 10_13_6) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/69.0.3497.81 Safari/537.36"},
    "Referer":         {"https://cn.pornhub.com/view_video.php?viewkey=ph5e35bb9dac481"},
    "Accept":          {"text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8"},
    "Accept-Charset":  {"UTF-8,*;q=0.5"},
    "Accept-Encoding": {"gzip,deflate,sdch"},
    "Accept-Language": {"en-US,en;q=0.8"},
}
Status Code: 200
panic: runtime error: index out of range [0] with length 0

goroutine 1 [running]:
github.com/iawia002/annie/downloader.Download(0x555dfb478cae, 0x13, 0xc00048ad31, 0x2d, 0x555dfb4709d4, 0x5, 0xc0000a63f0, 0x0, 0x0, 0x0, ...)
        github.com/iawia002/annie/downloader/downloader.go:196 +0x180e
main.download(0x7ffcd4205260, 0x3d, 0x7ffcd4205260)
        github.com/iawia002/annie/main.go:184 +0x400
main.main()
        github.com/iawia002/annie/main.go:238 +0x184
```

### PR DESCRIPTION
**Pull request by howeyc**
Need to run javascript in order to get the media urls.

Fixes #663 and #540

### CODE REVIEW

File: extractors/pornhub/pornhub.go [hunk 1]
```diff
@@ -48,8 +51,26 @@ func (e *extractor) Extract(url string, option types.Options) ([]*types.Data, er
...
+			}
+		}
+	}
+	vm := otto.New()
+	vm.Run(flashvar_js)
+	var urls []string
+	for i := 0; i < len(pornhubs); i++ {
+		if value, err := vm.Get(fmt.Sprintf("media_%d", i)); err == nil {
+			urls = append(urls, value.String())
+	        }
```
Comment (iawia002) [inline extractors/pornhub/pornhub.go:69]: gofmt here

File: extractors/pornhub/pornhub.go [hunk 2]
```diff
@@ -48,8 +51,26 @@ func (e *extractor) Extract(url string, option types.Options) ([]*types.Data, er
 		return nil, err
 	}
 
+	scripts := utils.MatchAll(html, `<script type="text/javascript">(?s)(.+?)</script>`)
+	var flashvar_js string
```
Comment (iawia002) [inline extractors/pornhub/pornhub.go:55]: better be `flashvarJS`

**Comment by howeyc** [comment]
Updated based on your comments

File: extractors/pornhub/pornhub.go [hunk 3]
```diff
@@ -48,9 +51,28 @@ func (e *extractor) Extract(url string, option types.Options) ([]*types.Data, er
...
+	for _, s := range scripts {
+		for _, js := range s[1:] {
+			if strings.Contains(js, "flashvars_") {
+				flashvarJS = js
+				break
+			}
+		}
+	}
+	vm := otto.New()
+	vm.Run(flashvarJS)
```
Comment (iawia002) [inline extractors/pornhub/pornhub.go:65]:
```suggestion
	vm.Run(flashvarJS) // nolint
```

One more thing, the error return value of `vm.Run` is not checked, this didn't pass golangci-lint's check

**Comment by howeyc** [comment]
Updated based on comments.

**Comment by Croddock** [comment]
> 需要运行 javascript 才能获取媒体网址。
> 
> 修复 #663和#540

请问具体的实现方法是什么？能否指导一下，谢谢

**Pull request merged**